Sources/OAuth2Client/NXOAuth2Connection.m

```diff
@@ -418,7 +418,7 @@ - (void)connection:(NSURLConnection *)connection didReceiveResponse:(NSURLRespon
     if (self.statusCode == 401
         && client.accessToken.refreshToken != nil
         && authenticateHeader
-        && [authenticateHeader rangeOfString:@"expired_token"].location != NSNotFound) {
+        && ([authenticateHeader rangeOfString:@"invalid_token"].location != NSNotFound || [authenticateHeader rangeOfString:@"invalid_request"].location != NSNotFound)) {
```

Comment:
This does not look right: The case is a 401 with expiration explicitly because we want to refresh the token in that case. The other cases need to be covered somewhere else

Also this is not part of the crash fix as well.